### PR TITLE
[SPARK-29902][Doc][Minor]Add listener event queue capacity configuration to documentation

### PR DESCRIPTION
**Pull request by shahidki31**
### What changes were proposed in this pull request?

Add listener event queue capacity configuration to documentation
### Why are the changes needed?

We some time see many event drops happening in eventLog listener queue. So, instead of increasing all the queues size, using this config we just need to increase eventLog queue capacity.

```
scala> sc.parallelize(1 to 100000, 100000).count()
[Stage 0:=================================================>(98299 + 4) / 100000]19/11/14 20:56:35 ERROR AsyncEventQueue: Dropping event from queue eventLog. This likely means one of the listeners is too slow and cannot keep up with the rate at which tasks are being started by the scheduler.
19/11/14 20:56:35 WARN AsyncEventQueue: Dropped 1 events from eventLog since the application started.
```

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

Existing tests

### CODE REVIEW

**Comment by shahidki31** [comment]
cc @srowen Kindly review

File: docs/configuration.md [hunk 1]
```diff
@@ -1857,6 +1857,61 @@ Apart from these, the following properties are also available, and may be useful
...
+    shared event queue hold events for external listener that register to listener bus.
+    Consider increasing value if listener events corresponding to shared queue are dropped.
+    Increasing this value may result in the driver using more memory.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.scheduler.listenerbus.eventqueue.appStatus.capacity</code></td>
+  <td><code>spark.scheduler.listenerbus.eventqueue.capacity</code></td>
+  <td>
+    Capacity for appStatus event queue in Spark listener bus, must be greater than 0.
```
Comment (srowen) [inline docs/configuration.md]: "appStatus event queue hold events for internal application status listeners" is useful;  "Capacity for appStatus event queue in Spark listener bus" seems sort of redundant. I'd maybe lead with "Capacity of the appStatus event queue, which holds events ..." just once. Here and elsewhere.

File: docs/configuration.md [hunk 2]
```diff
@@ -1857,6 +1857,61 @@ Apart from these, the following properties are also available, and may be useful
     driver using more memory.
   </td>
 </tr>
+<tr>
+  <td><code>spark.scheduler.listenerbus.eventqueue.shared.capacity</code></td>
+  <td><code>spark.scheduler.listenerbus.eventqueue.capacity</code></td>
+  <td>
+    Capacity for shared event queue in Spark listener bus, must be greater than 0.
+    Unless otherwise specified it uses <code>spark.scheduler.listenerbus.eventqueue.capacity</code>.
```
Comment (srowen) [inline docs/configuration.md]: I think this is covered already by the fact that this is shown as the default here. You can remove it everywhere.

**Review by srowen — CHANGES_REQUESTED** [review]

File: docs/configuration.md (hunk 1 above)
Comment (shahidki31) [reply docs/configuration.md]: Thanks @srowen , I have updated

File: docs/configuration.md (hunk 2 above)
Comment (shahidki31) [reply docs/configuration.md]: Done

**Comment by shahidki31** [comment]
Retest this please

**Comment by shahidki31** [comment]
retest this please

**Comment by shahidki31** [comment]
I am not sure why the checks are failing. Seems unrelated error?

**Comment by srowen** [comment]
You can ignore the checks below at the moment. It is unrelated.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Comment by shahidki31** [comment]
Thanks @srowen

**Comment by jiangxb1987** [comment]
https://github.com/apache/spark/pull/25307#issuecomment-517473871
Is it safe to mention the queue names in the document?

**Comment by srowen** [comment]
Hm, good question. They're not explicitly marked internal, and I don't expect (?) they'll change. At the least maybe this could carry a comment that the names of the queues are subject to change.

**Comment by shahidki** [comment]
@jiangxb1987 I think from the console/logs , user can know exactly from which queue event drop has happened and how many events dropped from that particular queue. Also user can see which event has dropped, if enable the trace.

**Pull request closed without merging**